### PR TITLE
platform(general): Get resources from platform and filter taggable resources for policies

### PR DESCRIPTION
**Pull request by mikeurbanski1**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

- Gets the list of all resource types and their attributes from the platform
- Also gets a list of attributes that we want to filter by resource type (this is just tags / labels for the various providers right now)
- Gets the provider from the updated platform policy metadata response
- Checkov will apply the logic to replace any policies using these attributes with a resource type of "all" with the resource types for the given attribute and provider (if available).

So, for example, if you have a platform policy with provider `aws` and a condition that checks `tags` with resource types `all`, then the check will only apply to taggable AWS Terraform resources. If you use the attribute `Tags.*.Key`, then it will only apply to CloudFormation.

If you have a Checkov policy that uses `all` with the attribute `tags`, then it will apply it to AWS, Azure, and Ali (because checkov policies do not have a provider in the spec).

This also removes the local list of resource types from checkov, which was pretty much totally dead code in reality, because platform custom policies didn't even have the provider in the API response.

Requires some platform changes (some done, some in progress) before this can be 100% safely merged (but it _should_ still work)

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [X] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mikeurbanski1** [comment]
I am in typing hell

File: checkov/common/bridgecrew/integration_features/features/attribute_resource_types_integration.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+if TYPE_CHECKING:
+    from checkov.common.bridgecrew.platform_integration import BcPlatformIntegration
+
+
+ALL_TYPES = '__all__'
+
+
+class AttributeResourceTypesIntegration(BaseIntegrationFeature):
+    def __init__(self, bc_integration: BcPlatformIntegration) -> None:
+        super().__init__(bc_integration=bc_integration, order=1)  # must be after policy metadata
```
Comment (tronxd) [inline checkov/common/bridgecrew/integration_features/features/attribute_resource_types_integration.py:20]: `CustomPoliciesIntegration` order's is already 1. Can we assign a unique order?

File: checkov/common/bridgecrew/integration_features/features/attribute_resource_types_integration.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        if not attr:
+            return None
+        if '.' in attr:
+            attr = attr[0:attr.index('.')]
+
+        resource_types = self.attribute_resources.get(attr, None)
+        if not resource_types:
+            return None
+
+        return resource_types.get(provider or '__all__')
```
Comment (tronxd) [inline checkov/common/bridgecrew/integration_features/features/attribute_resource_types_integration.py:69]:
```suggestion
        return resource_types.get(provider or ALL_TYPES)
```

File: checkov/common/bridgecrew/integration_features/features/attribute_resource_types_integration.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    def _build_attribute_resource_map(self, resource_definitions: _ResourceDefinitions) -> None:
```
Comment (tronxd) [inline checkov/common/bridgecrew/integration_features/features/attribute_resource_types_integration.py:71]: Can we add a pydoc to the function?

File: checkov/common/checks_infra/checks_parser.py [hunk 4]
```diff
@@ -133,18 +134,33 @@
...
-        check.id = raw_check.get("metadata", {}).get("id", "")
-        check.name = raw_check.get("metadata", {}).get("name", "")
-        check.category = raw_check.get("metadata", {}).get("category", "")
-        check.frameworks = raw_check.get("metadata", {}).get("frameworks", [])
-        check.guideline = raw_check.get("metadata", {}).get("guideline")
+
+        metadata = raw_check.get("metadata", {})
+
+        # the first approach comes from the custom policy integration
+        provider = metadata.get("scope", {}).get("provider")\
```
Comment (tronxd) [inline checkov/common/checks_infra/checks_parser.py:141]:
```suggestion
        provider = metadata.get("scope", {}).get("provider")
```

File: checkov/common/checks_infra/solvers/attribute_solvers/base_attribute_solver.py [hunk 5]
```diff
@@ -85,7 +85,11 @@ def get_attribute_matches(self, vertex: Dict[str, Any]) -> List[str]:
         if self.is_jsonpath_check:
             parsed_attr = self.parsed_attributes.get(self.attribute)
             if parsed_attr is None:
-                parsed_attr = parse(self.attribute)
+                try:
+                    parsed_attr = parse(self.attribute)
+                except Exception as e:
+                    print(e)
```
Comment (tronxd) [inline checkov/common/checks_infra/solvers/attribute_solvers/base_attribute_solver.py:91]:
```suggestion
                    logging.error(e)
```

**Review by tronxd — APPROVED** [review]
LGTM, had some comments

**Comment by mikeurbanski1** [comment]
@gruebel @nimrodkor @rotemavni added you to sanity check whether this can be safely merged. I ran some tests against prod (where the platform changes do not exist) and it seems to work fine in checkov. And I don't think anything here will break a platform scan in the checkov wrapper.

**Review by gruebel — APPROVED** [review]
nice, great work. The only question I had, you already answered in the description 🥇

**Pull request merged**